Gems/LyShine/Code/Source/UiCanvasManager.cpp

```diff
@@ -805,94 +808,98 @@ UiCanvasComponent* UiCanvasManager::FindEditorCanvasComponentByPathname(const AZ
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
-bool UiCanvasManager::HandleInputEventForInWorldCanvases([[maybe_unused]] const AzFramework::InputChannel::Snapshot& inputSnapshot, [[maybe_unused]] const AZ::Vector2& viewportPos)
+bool UiCanvasManager::HandleInputEventForInWorldCanvases(const AzFramework::InputChannel::Snapshot& inputSnapshot, const AZ::Vector2& viewportPos)
 {
-    // First we need to construct a ray from the either the center of the screen or the mouse position.
-    // This requires knowledge of the camera
-    // for initial testing we will just use a ray in the center of the viewport
+    if (!AZ::Interface<ILyShine>::Get())
+    {
+        return false;
+    }
+
+    // First we need to construct a ray from either the center of the screen or the mouse position
+    CLyShine* lyShine = static_cast<CLyShine*>(AZ::Interface<ILyShine>::Get());
```

Comment:
Is this a common pattern? We might want to add an inline helper to `CLyShine` or expand `ILyShine` to contain the implementation details you need here.

Reply:
If you're referring to the static_cast, then not really. It's because there's currently no IUiRenderer interface. I've added a ticket to address this.